src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -51,18 +52,26 @@ internal abstract class AbstractSnippetInfoService : ForegroundThreadAffinitized
             IAsynchronousOperationListenerProvider listenerProvider)
             : base(threadingContext)
         {
-            AssertIsForeground();
-
             if (serviceProvider != null)
             {
-                var textManager = (IVsTextManager2)serviceProvider.GetService(typeof(SVsTextManager));
-                if (textManager.GetExpansionManager(out _expansionManager) == VSConstants.S_OK)
-                {
-                    ComEventSink.Advise<IVsExpansionEvents>(_expansionManager, this);
-                    _waiter = listenerProvider.GetListener(FeatureAttribute.Snippets);
-                    _languageGuidForSnippets = languageGuidForSnippets;
-                    PopulateSnippetCaches();
-                }
+                _waiter = listenerProvider.GetListener(FeatureAttribute.Snippets);
+                _languageGuidForSnippets = languageGuidForSnippets;
+
+                InitializeAndPopulateSnippetsCacheAsync(threadingContext, serviceProvider).Forget();
```

Comment:
💭 We could really use a `FileAndForget` method (related to #29273)

Reply:
There is one I can import from the VS shell - https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.vstasklibraryhelper.fileandforget?view=visualstudiosdk-2019

Should I use this instead?

Reply:
Ours should tie into the Roslyn telemetry/NFW infrastructure so we know things are operating in a consistent manner.

Reply:
So what may be better than the true fire-and-forget is to instead add the task into a JointableTaskCollection and wait for that when this MEF component is Disposed; that way this isn't going to risk running unsynchronized during VS shutdown.

Reply:
We do this in other places too for stuff written since we tried to be good citizens on this.

Reply:
> 
> 
> So what may be better than the true fire-and-forget is to instead add the task into a JointableTaskCollection and wait for that when this MEF component is Disposed; that way this isn't going to risk running unsynchronized during VS shutdown.

I'm not sure I quite understand this comment - do you mean similar to https://github.com/dotnet/roslyn/blob/9625a14e7405d7a3ea1dc77957d5293850b69cc8/src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs#L42 ?

It seems a little tricky as I'll need to deal with multiple tasks in that collection since it is not just called once from the ctor.